proposals/0132-sequence-end-ops.md

```diff
@@ -16,724 +17,227 @@ these names so they follow consistent, predictable patterns.
-don't know what its inverse is, you will probably not guess that it's 
-`dropFirst(10)`. The confusing, conflicting names in these APIs are a 
-barrier to users adopting them where appropriate.
+  [onesided]: (0172-one-sided-ranges.md)
+  
+| Operation        | Operand                           | Current name                  |
+| ---------------- | --------------------------------- | ----------------------------- |
+| Get value        | First element                     | `first`                       |
+| "                | Last element                      | `last`                        |
+| "                | First *n* elements                | `prefix(3)`                   |
+| "                | Last *n* elements                 | `suffix(3)`                   |
+| "                | Leading elements matching closure | `prefix(while: isOdd)`        |
+| "                | Earliest element matching closure | `first(where: isOdd)`         |
+| Get index        | Earliest element equal to value   | `index(of: x)`                |
+| "                | Earliest element matching closure | `index(where: isOdd)`         |
+| Remove from copy | First element                     | `dropFirst()`                 |
+| "                | Last element                      | `dropLast()`                  |
+| "                | First *n* elements                | `dropFirst(3)`                |
+| "                | Last *n* elements                 | `dropLast(3)`                 |
+| "                | Leading elements matching closure | `drop(while:)`                |
+| Remove from self | First element                     | `removeFirst()`               |
+| "                | Last element                      | `removeLast()`                |
+| "                | First *n* elements                | `removeFirst(3)`              |
+| "                | Last *n* elements                 | `removeLast(3)`               |
+| …if present      | First element                     | `popFirst()`                  |
+| "                | Last element                      | `popLast()`                   |
+| Test equality    | First *n* elements                | `starts(with: other)`         |
+| …with function   | "                                 | `starts(with: other, by: ==)` |
+
+Put next to each other, we see a lot of inconsistent terminology:
+
+* Usually, "first *n* elements" is handled by overloading a `first` method, except 
+  on the most-used category, "get value of element(s)". There, we suddenly 
+  use `prefix` and `suffix`.
+
+* The "get index" methods do not indicate a direction, but adding 
+  versions which search from the end would be very plausible. Similarly, 
+  `drop(while:)` does not include a direction, but dropping trailing elements is a plausible feature.
+
+* "Remove from copy" and "Remove from self" are 
+  closely related, but they have unrelated names. The name `drop`, while a 
```

Comment:
I strongly disagree that these operations are closely related.  One is a category of slicing ops that implies no copying, the other is a mutating operation that may very well need to copy elements.

Reply:
I lost the first version of this comment, but tl;dr is, I think these are best modeled as a mutating/nonmutating pair. Doing so makes these operations clear at the point of use and discoverable in documentation.